src/shogun/optimization/DescendCorrection.h

```diff
@@ -68,11 +68,12 @@ class DescendCorrection
 	 *
 	 * @param negative_descend_direction the negative descend direction
 	 * @param idx the index of the direction
+	 * @param delta update the change to correct descend direction
 	 * 
 	 * @return corrected descend direction
 	 */
 	virtual float64_t get_corrected_descend_direction(float64_t negative_descend_direction,
-		index_t idx)=0;
+		index_t idx, float64_t& delta)=0;
```

Comment:
Yes.
If we allow to return a `vector`, we do not need to use reference.
However, the `delta` is not a `corrected_descend_direction`.